### DISCUSSION TITLE
How to get remainingTime of task in v2 ?

### DISCUSSION THREAD

**Discussion by klschoef** [lucaro/DRES#456] [Q&A] [answered]
Previously we could query the server for the remaining time of a task, and whether it was running? How to do this with getApiV2ClientEvaluationCurrentTaskByEvaluationId() and the returned ApiTaskTemplateInfo ?

**Comment by sauterl** [comment, marked as answer]
Hi there, thanks for the question.

Indeed, the `api/v2/client/evaluation/currentTask/{evaluationId}` endpoint only provides information on the current task _template_ as a feature.

In v2 this information is part of the state of the evaluation, hence the endpoint `/api/v2/evaluation/state/list` provides this information in the field `timeLeft`. Please note that the fields `timeElapsed` and `timeLeft` are seconds. Its sum might be not what the _template_ suggests, as administrators can modify the time of a task on-the-fly.

Since you provided Java / Kotlin names, please use `getApiV2EvaluationStateList`, which returns an array of `ApiEvaluationState` objects.

Generally speaking, we provide for your convenience the OpenApi specification in a human readable format, the [swagger editor](https://editor.swagger.io/?url=https://raw.githubusercontent.com/dres-dev/DRES/master/doc/oas-client.json) (this is the same link as the in-editor badge in our readme). However there one cannot try out those examples as there is no DRES instance. If enabled, this documentation can also be accessed by `<host>:<port>/swagger-client` in a browser.

**Reply by klschoef** [reply]
Great, thank you very much for the quick clarification!
